### PR TITLE
32-bit support

### PR DESCRIPTION
**Pull request by Goldfish64**
Support for macOS 10.4 and other 32-bit versions

- [x] ALCUserClient::externalMethod unavailable on 10.4 - **resolved in https://github.com/acidanthera/AppleALC/commit/1001ce20e505f055b44039969d57e0bb9e5b3823**
- [x] Add STAC9200 patches for 10.6 and older
- [x] Add STAC9205 for Dell Inspiron 1520
- [x] Fix 32-bit CI

### CODE REVIEW

File: AppleALC.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -1132,7 +1289,7 @@
 				CODE_SIGN_IDENTITY = "";
 				COPY_PHASE_STRIP = NO;
 				CURRENT_PROJECT_VERSION = "$(MODULE_VERSION)";
-				DEPLOYMENT_POSTPROCESSING = YES;
+				DEPLOYMENT_POSTPROCESSING = NO;
```
Comment (vit9696) [inline AppleALC.xcodeproj/project.pbxproj:1292]: Do these disable stripping for all archs? Might be not nice for file size.

File: AppleALC/AppleALC-Info.plist [hunk 2]
```diff
@@ -62,17 +62,15 @@
 		<key>com.apple.iokit.IOPCIFamily</key>
 		<string>1.0.0b1</string>
 		<key>com.apple.kpi.bsd</key>
-		<string>10.0.0</string>
-		<key>com.apple.kpi.dsep</key>
```
Comment (vit9696) [inline AppleALC/AppleALC-Info.plist:66]: is dsep really optional? Can you check with kextlib utility? It is built into macOS.

File: AppleALC/kern_alc.cpp [hunk 3]
```diff
@@ -798,7 +813,25 @@ IOReturn AlcEnabler::initializePinConfig(IOService *hdaCodec, IOService *configD
...
+	while (parentDevice) {
+		auto name = parentDevice->getName();
+		if (name && (!strcmp(name, "AppleHDAController"))) {
+			break;
+		}
 
+		parentDevice = parentDevice->getParentEntry(gIOServicePlane);
+	}
+
+	callbackAlc->patchPinConfig(hdaCodec, parentDevice);
```
Comment (vit9696) [inline AppleALC/kern_alc.cpp:829]: Check parentDevice for NULL here.

File: Resources/STAC9205/Info.plist [hunk 4]
```diff
@@ -0,0 +1,364 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (vit9696) [inline Resources/STAC9205/Info.plist:1]: Strip the xml tags like everywhere else.

File: AppleALC/AppleALC-Info.plist (hunk 2 above)
Comment (Goldfish64) [reply AppleALC/AppleALC-Info.plist:66]: dsep is not present in 10.4, and is not directly used within this plugin.

File: AppleALC.xcodeproj/project.pbxproj (hunk 1 above)
Comment (Goldfish64) [reply AppleALC.xcodeproj/project.pbxproj:1292]: Added equivalent strip command in https://github.com/acidanthera/AppleALC/commit/41aff415890a44c76909cc487bc37f4c555e4617. Deployment postprocessing must be disabled to ensure the binary is not modified after scripts are run.

File: AppleALC/kern_alc.cpp (hunk 3 above)
Comment (Goldfish64) [reply AppleALC/kern_alc.cpp]: Fixed in https://github.com/acidanthera/AppleALC/commit/b1047c1a3323c4759b9fc1ee5392ba20bc26b75d.

**Pull request merged**